### ISSUE TITLE
Replace Zeroizing derive(Debug) with an empty block

### ISSUE THREAD

**Issue by kayabaNerve** [RustCrypto/utils#825]
I do understand this is opinionated, yet if I want the memory cleared when I'm done with it, I don't want to hear it's still in a log file somewhere. In my opinion, at least by default, `Zeroizing` should stub `Debug` with:

```rs
use core::fmt::{Debug, Formatter};
impl<Z> Debug for Zeroizing<Z> {
  fn fmt(&self, fmt: &mut Formatter<'_>) -> Result<(), core::fmt::Error> {
    fmt.debug_struct("Zeroizing").finish()
  }
}
```

**Comment by tarcieri** [comment]
Yeah, that would be OK, although it might be helpful to still display the inner type name via `DebugStruct` (as sourced via `core::any::type_name`) + `finish_non_exhaustive`

**Comment by kayabaNerve** [comment]
Thanks for the tip on type_name. I learned of finish_non_exhaustive after this, yet can make a PR with this later :+1:

**Comment by newpavlov** [comment]
I don't think it's an area of responsibility for `zeroize`. Arguably, you should use the [`opaque-debug`](https://docs.rs/opaque-debug) crate or add a manual opaque `Debug` implementation for your type which gets wrapped by `Zeroizing`.

**Comment by Pandapip1** [comment]
> I don't think it's an area of responsibility for zeroize. Arguably, you should use the [opaque-debug](https://docs.rs/opaque-debug) crate or add a manual opaque Debug implementation for your type which gets wrapped by Zeroizing.

Okay, but if that's the case then Zeroizing shouldn't implement Debug at all then. If I submitted a PR that added this as an optional feature, would it likely be accepted?

**Comment by tarcieri** [comment]
Please don't add additional dependencies to `zeroize`, especially for something as trivial as `opaque-debug`

**Comment by Pandapip1** [comment]
From `opaque-debug`:

> This crate is deprecated. Use explicit impl of fmt::Debug instead.

I'm not suggesting to add additional dependencies, I'm suggesting that the (optional!) feature be added.
